data-prepper-plugins/http-source/src/main/java/com/amazon/dataprepper/plugins/source/loghttp/HTTPSource.java

```diff
@@ -32,7 +32,7 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ScheduledThreadPoolExecutor;
 
-@DataPrepperPlugin(name = "http", pluginType = Source.class)
+@DataPrepperPlugin(name = "http", pluginType = Source.class, pluginConfigurationType = HTTPSourceConfig.class)
```

Comment:
QUES: 

(1) What will happen if the pluginType does not match the implemented interface?
(2) Looks like this introduces two sources of truth for pluginType. What is the long term plan?

Reply:
If `pluginType` does not match the implemented interface, then the plugin framework will throw an exception at runtime. I added the `pluginType` in another PR a few weeks ago. I opted for the approach of having the `pluginType` and the interface because I want plugins to be able to implement other interfaces if they find appropriate.

Reply:
Thanks. I missed that part.

> I opted for the approach of having the pluginType and the interface because I want plugins to be able to implement other interfaces if they find appropriate.

Could you provide a toy example for this use case? b/c I am still confused how one can avoid `implements Source<Record<String>>` since it involves the data type claim.

Reply:
Oh. Wait. My mis. Java supports multiple interface but not multiple inheritance.